File: src/Squad.Client/src/lib/interactions.js
// Activity social layer — kudos + comments. Thin wrappers over the squad-scoped
// endpoints under /api/activities/{id}. All take the bearer token (the caller
// resolves it via getToken); the global fetch shim prefixes /api on native.

const authHeaders = (token) => (token ? { Authorization: `Bearer ${token}` } : undefined);

// Toggle kudos on an activity. give=true → POST, give=false → DELETE.
// Returns the fresh { count, kudoed } state.
export async function setKudos(activityId, give, token) {
  const res = await fetch(`/api/activities/${activityId}/kudos`, {
    method: give ? 'POST' : 'DELETE',
    headers: authHeaders(token),
  });
  if (!res.ok) throw new Error(`Kudos failed (${res.status})`);
  return res.json();
}

// The activity's comment thread, oldest-first (enriched with author display fields).
export async function fetchComments(activityId, token) {
  const res = await fetch(`/api/activities/${activityId}/comments`, { headers: authHeaders(token) });
  if (!res.ok) throw new Error(`Comments failed (${res.status})`);
  return res.json();
}

// Post a comment; returns the created (enriched) comment.
export async function postComment(activityId, body, token) {
  const res = await fetch(`/api/activities/${activityId}/comments`, {
    method: 'POST',
    headers: { ...(authHeaders(token) || {}), 'Content-Type': 'application/json' },
    body: JSON.stringify({ body }),
  });
  if (!res.ok) throw new Error(`Post comment failed (${res.status})`);
  return res.json();
}

File: src/Squad.Core/Contracts/Interactions.cs
namespace Squad.Core;

// ----- Activity interactions: kudos + comments -----
// Strava-style social layer over a committed activity. Both are squad-scoped: an
// athlete can only react to / comment on activities owned by a member of their squad.

/// <summary>An activity's kudos state for one caller: total count + whether they've kudoed it.</summary>
public sealed record KudosState(int Count, bool Kudoed);

/// <summary>A comment on an activity, enriched with the author's display fields (rendered directly).</summary>
public sealed record ActivityComment(
    Guid Id, Guid ActivityId, Guid AthleteId, string AthleteName, string Initials, string AvatarColor,
    string Body, DateTimeOffset CreatedUtc,
    // Proxy path to the author's avatar photo (null when they have none → initials).
    string? AvatarUrl = null);

/// <summary>Give/remove kudos on a squad activity. Returns null when the activity isn't visible
/// in the caller's squad; otherwise the fresh count + whether the caller now has kudos on it.</summary>
public interface IKudosService
{
    Task<KudosState?> SetAsync(Guid activityId, Guid squadId, Guid athleteId, bool give, CancellationToken ct);
}

/// <summary>Read/post comments on a squad activity. A null result means the activity isn't
/// visible in the caller's squad (never a member's activity → 404 at the edge).</summary>
public interface ICommentService
{
    Task<IReadOnlyList<ActivityComment>?> GetAsync(Guid activityId, Guid squadId, int take, CancellationToken ct);
    /// <summary>Persist a comment and return it enriched with the author's display fields.</summary>
    Task<ActivityComment?> PostAsync(Guid activityId, Guid squadId, Guid athleteId, string body, CancellationToken ct);
}
